dask/array/core.py

```diff
@@ -1156,6 +1156,24 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
             if result is not None:
                 self = result
 
+        try:
+            layer = self.dask.layers[name]
+        except (AttributeError, KeyError):
+            # self is no longer an Array after applying the plugins, OR
+            # a plugin replaced the HighLevelGraph with a plain dict, OR
+            # name is not the top layer's name (this can happen after the layer is
+            # manipulated, to avoid a collision)
+            pass
+        else:
+            layer.collection_annotations.update(
+                {
+                    "type": type(self),
+                    "chunk_type": type(meta),
+                    "chunks": chunks,
+                    "dtype": dtype,
```

Comment:
Similar comment here for using `self.chunks` and `self.dtype`. I'll defer to you about what you think is best to keep around for visualization purposes. Though note that while `chunks` can be things like `"auto"` and `2`, `self.chunks` will be the expanded chunks for the full array like `((2, 2, 2, 2, 2),)`.

Reply:
There are several cases where using `self.dtype` like you suggest produces failing tests in `dask/test/test_gufunc.py`

These tests fail because in these cases `meta` / `self._meta` is a tuple. Tuples do not have a `.dtype` attribute, so when we look up `self.dtype` we get an error. 

```python
    @property
    def dtype(self):
>       return self._meta.dtype
```

```
dask/array/tests/test_gufunc.py::test_apply_gufunc_01 FAILED             [ 12%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_01b FAILED            [ 14%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_output_dtypes_string_many_outputs[False] FAILED [ 20%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_output_dtypes_string_many_outputs[True] FAILED [ 22%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_elemwise_02 FAILED    [ 34%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_elemwise_loop FAILED  [ 38%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_elemwise_core FAILED  [ 40%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_two_scalar_output FAILED [ 42%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_two_mixed_outputs FAILED [ 44%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_broadcasting_loopdims FAILED [ 58%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_infer_dtype FAILED    [ 66%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_axis_02 FAILED        [ 72%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_axis_02b FAILED       [ 74%]
dask/array/tests/test_gufunc.py::test_preserve_meta_type FAILED          [ 96%]
dask/array/tests/test_gufunc.py::test_apply_gufunc_with_meta FAILED      [ 98%]
dask/array/tests/test_gufunc.py::test_as_gufunc_with_meta FAILED         [100%]
```

There are no problems with `self.chunks`, and in fact it's preferable to have the full expanded chunks listed instead of `"auto"`.

Reply:
Thanks for looking into this @GenevieveBuckley. Let's use `dtype` then -- we can always make updates as needed later on